### ISSUE TITLE
add new member to etcd, and etcd start failed

### ISSUE THREAD

**Issue by tbchj** [coreos/etcd#7820]
before I  extended my etcd cluster, I had only one member, and then I want to extend it to three member, but when I use etcdctl member add ...  one member , and etcd service failed. and I can't start it again.
I know the reason why it can't start.
the cluster now has two members, and can't election successfully, but what can I do? except remove all the data directory, how to rollback or how to go ahead?
the detail log is here:   https://github.com/tbchj/etcd-error
Is there any bugs ? if cluster only and just have  some couples of member, how to do ? like the situation I faced.

**Comment by gyuho** [comment]
When you add a new member to a single node cluster, the quorum becomes 2 (=2/2+1). So a candidate needs votes from all of those two members to become the leader.

Your logs show:

```
Apr 26 23:15:56  etcd: 3ae830665dc1d868 received vote from 3ae830665dc1d868 at term 28
Apr 26 23:15:56  etcd: 3ae830665dc1d868 [logterm: 2, index: 2137] sent vote request to 39d23979d4ab4774 at term 28
Apr 26 23:15:57  etcd: 3ae830665dc1d868 is starting a new election at term 28
Apr 26 23:15:57  etcd: 3ae830665dc1d868 became candidate at term 29
Apr 26 23:15:57  etcd: 3ae830665dc1d868 received vote from 3ae830665dc1d868 at term 29
Apr 26 23:15:57  etcd: 3ae830665dc1d868 [logterm: 2, index: 2137] sent vote request to 39d23979d4ab4774 at term 29
Apr 26 23:15:59  etcd: 3ae830665dc1d868 is starting a new election at term 29
Apr 26 23:15:59  etcd: 3ae830665dc1d868 became candidate at term 30
Apr 26 23:15:59  etcd: 3ae830665dc1d868 received vote from 3ae830665dc1d868 at term 30
Apr 26 23:15:59  etcd: 3ae830665dc1d868 [logterm: 2, index: 2137] sent vote request to 39d23979d4ab4774 at term 30
Apr 26 23:16:00  etcd: 3ae830665dc1d868 is starting a new election at term 30
Apr 26 23:16:00  etcd: 3ae830665dc1d868 became candidate at term 31
```

`3ae830665dc1d868` keeps sending vote request but could not receive the response  from `39d23979d4ab4774` within the election timeout.

Maybe the round trip time between two machines is greater than your election timeout `1000ms`? Or do you have logs of the other node `39d23979d4ab4774`? It will be helpful to see what's going on in the whole cluster.

**Comment by xiang90** [comment]
@tbchj Did you start your 2nd etcd member?

**Comment by tbchj** [comment]
thanks for your reply.
sorry, I didin't  save the other node log, but I experenced again.
the  node which was added can't start successfully, always failed.
and I experenced two situations. one is : after adding the new member, the other member which already exist  in the etcd cluster, the service status is  normal (service status is activing).
another situation is : after adding new member, the already exist member service status is not normal(service status is dead)   
but both all of them , the added member can't start successfully. the added member log is : https://github.com/tbchj/etcd-error/blob/master/etcd-start.log (situation 1)  and https://github.com/tbchj/etcd-error/blob/master/etcd-stop.log (situation 2).
3ae830665dc1d868  is the original member， 8154da47cc70c725  like 39d23979d4ab4774(above) is the added member

**Comment by tbchj** [comment]
maybe I know how to solve the problem. when I asked the question yesterday, I had no time to experence because someone must use the cluster(k8s) to test some work. 
yesterday,  after I saw this below:
```
[root@ ]# etcdctl member list
Failed to get leader:  client: etcd cluster is unavailable or misconfigured
```
I just thought the original member etcd service dead. add I start the added member,but failed again. and the worsest is I didin't save the log, but I remembered the service was  not normal.

I just tried again, after original etcd add member, both two member service dead, but I cleaned  the  added member data directory(/var/lib/etcd) , add I  started both two member etcd service, and it works, now my cluster just has two member, one leader one follower. maybe the question solved, and I think etcd cluster will facing couples of member many times in reality.

**Comment by gyuho** [comment]
@tbchj Can you share the exact steps to reproduce your issue?

The existing member experiences errors when the quorum is lost (e.g. add 1 member to single node cluster). That's why you need to start the new member right after you run `member add` command.

**Comment by tbchj** [comment]
sorry, the first time I extended my one member cluster  did failed, and at that time,  the network of my cluster  is unstable,  after I using "etcdctl member add ..." add one member, then I start the new member , maybe because of the factor of network , the new member service start failed. I checked the log of new member, just failed. 
maybe the experence I tried yesterday, the situation not the same because of many factors. but I finally know how to add new member and how to solve the question after trying many times, thank you.
